### ISSUE TITLE
English: $ERROR_INFO doesn't refer to the last error that was raised

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1660]
This construct is used in `dry-monads` specs https://github.com/dry-rb/dry-monads/blob/master/spec/unit/try_spec.rb#L14

`$ERROR_INFO` is alias of `$!`

TruffleRuby 1.0.0 RC15

```ruby
require 'English'
division_error = 1 / 0 rescue $ERROR_INFO
# => nil
```

MRI 2.6.2

```ruby
require 'English'
division_error = 1 / 0 rescue $ERROR_INFO
# => #<ZeroDivisionError: divided by 0>
```

**Comment by eregon** [comment]
I thought this might relate to #1560 but apparently not:
```ruby
$ irb
irb(main):001:0> require 'English'
=> true
irb(main):002:0> division_error = 1 / 0 rescue $ERROR_INFO
=> nil
irb(main):003:0> division_error = 1 / 0 rescue $!
=> #<ZeroDivisionError: divided by 0>
```

**Comment by eregon** [comment]
Probably, the `$ERROR_INFO` alias doesn't work because `$!` is thread-local, but I need to investigate.
Thanks for the report.

**Comment by deepj** [comment]
@eregon any news about this? :)

**Comment by bjfish** [comment]
@deepj This is fixed at 6c34ed26d522bbfc47e4fc99963adc57ae9c362f. Thanks!
